manifests/init.pp

```diff
@@ -100,7 +100,8 @@
   Optional[String] $systemd_beat_log_opts_override                    = undef,
   String $systemd_beat_log_opts_template                              = $filebeat::params::systemd_beat_log_opts_template,
   String $systemd_override_dir                                        = $filebeat::params::systemd_override_dir,
-
+  Variant[String, Enum['exec', 'chocolatey']] $package_provider       = $filebeat::params::package_provider,
+  Optional[String[1]] $package_name                                   = $filebeat::params::package_name,
```

Comment:
While a little out of scope from the original MR, if we used the `package_name` parameter on non-windows kernels we'd end up solving #179 too. I'm happy to make that change after merging this if you'd rather stick with this though (I completely understand).

Reply:
Well I look at #179  it seems quite complex to me. What I can do is to change openbsd.pp /linux.pp / openbsd.pp to use $package_name   and into params.pp manage default value for openbsd what do you think ?

Reply:
I think making the package name dynamic in openbsd.pp and linux.pp is enough. I think edge cases can be handled between a dynamic name and setting `manage_repo` to false.

Reply:
Done !